From f0c3006c96a8be366dde4fb0f0f2355eb002ea3d Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Mon, 18 Feb 2019 16:02:03 +0100
Subject: [PATCH] tqma8mq: add 2 GiB module variants

per default we will support 1 GiB LPDDR4 (not available at the moment)
support two variants with 2 GiB LPDDR4, handled per Kconfig settings

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 board/tqc/tqma8mx/Kconfig                     | 15 ++++--
 board/tqc/tqma8mx/spl.c                       | 13 ++++-
 configs/tqma8mq_2gmicron_mba8mx_mmc_defconfig | 47 +++++++++++++++++++
 include/configs/tqma8mx.h                     | 14 ++++--
 4 files changed, 79 insertions(+), 10 deletions(-)
 create mode 100644 configs/tqma8mq_2gmicron_mba8mx_mmc_defconfig

diff --git a/board/tqc/tqma8mx/Kconfig b/board/tqc/tqma8mx/Kconfig
index c0e355976e..b046395bc5 100644
--- a/board/tqc/tqma8mx/Kconfig
+++ b/board/tqc/tqma8mx/Kconfig
@@ -10,15 +10,20 @@ config SYS_CONFIG_NAME
 	default "tqma8mx"
 
 choice
-	prompt "TQMa8MX CPU variant"
-	default TQMA8MX_MX8MQ
+	prompt "TQMa8MX Memory variant"
+	default TQMA8MX_2G_SAMSUNG
 	help
 	  Select cpu variant for TQMa8MX
 
-config TQMA8MX_MX8MQ
-	bool "TQMa8MX with i.MX8MQ"
+config TQMA8MX_2G_SAMSUNG
+	bool "TQMa8MX with 2GB LPDDR4 Samsung"
 	help
-	  Select the CPU variant
+	  Select module with 2GB LPDDR4 Samsung"
+
+config TQMA8MX_2G_MICRON
+	bool "TQMa8MX with 2GB LPDDR4 Micron"
+	help
+	  Select module with 2GB LPDDR4 Micron"
 
 endchoice
 
diff --git a/board/tqc/tqma8mx/spl.c b/board/tqc/tqma8mx/spl.c
index 35594038f5..2e04e21e1f 100644
--- a/board/tqc/tqma8mx/spl.c
+++ b/board/tqc/tqma8mx/spl.c
@@ -24,13 +24,14 @@
 #include <asm/arch/imx8m_ddr.h>
 
 #include "../common/tqc_bb.h"
+#include "../common/tqc_eeprom.h"
 
 DECLARE_GLOBAL_DATA_PTR;
 
 extern struct dram_timing_info dram_timing_2gs;
 extern struct dram_timing_info dram_timing_2gm;
 
-void spl_dram_init(void)
+static void spl_dram_init(void)
 {
 	u32 rev = get_cpu_rev() & 0xfff;
 
@@ -40,13 +41,19 @@ void spl_dram_init(void)
 		printf("SPL: no timing for this chip rev\n");
 		hang();
 	} else {
+#if defined(CONFIG_TQMA8MX_2G_SAMSUNG)
 		ddr_init(&dram_timing_2gs);
+#elif defined(CONFIG_TQMA8MX_2G_MICRON)
+		ddr_init(&dram_timing_2gm);
+#else
+#error
+#endif
 	}
 }
 
 #define I2C_PAD_CTRL	(PAD_CTL_DSE6 | PAD_CTL_HYS | PAD_CTL_PUE)
 #define PC MUX_PAD_CTRL(I2C_PAD_CTRL)
-struct i2c_pads_info i2c_pad_info1 = {
+static struct i2c_pads_info i2c_pad_info1 = {
 	.scl = {
 		.i2c_mode = IMX8MQ_PAD_I2C1_SCL__I2C1_SCL | PC,
 		.gpio_mode = IMX8MQ_PAD_I2C1_SCL__GPIO5_IO14 | PC,
@@ -200,6 +207,8 @@ int board_fit_config_name_match(const char *name)
  */
 void board_init_f(ulong dummy)
 {
+	uchar rcw[32];
+	char *ram_type;
 	int ret;
 
 	/* Clear global data */
diff --git a/configs/tqma8mq_2gmicron_mba8mx_mmc_defconfig b/configs/tqma8mq_2gmicron_mba8mx_mmc_defconfig
new file mode 100644
index 0000000000..1f1eb61c55
--- /dev/null
+++ b/configs/tqma8mq_2gmicron_mba8mx_mmc_defconfig
@@ -0,0 +1,47 @@
+CONFIG_ARM=y
+CONFIG_ARCH_IMX8M=y
+CONFIG_SYS_TEXT_BASE=0x40200000
+CONFIG_SYS_MALLOC_F_LEN=0x2000
+CONFIG_TARGET_TQMA8MX=y
+CONFIG_TQMA8MX_2G_MICRON=y
+CONFIG_DEFAULT_DEVICE_TREE="fsl-imx8mq-tqma8mq-mba8mx"
+CONFIG_FIT=y
+CONFIG_SPL_LOAD_FIT=y
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=arch/arm/mach-imx/spl_sd.cfg"
+CONFIG_ARCH_MISC_INIT=y
+CONFIG_SPL=y
+CONFIG_SPL_BOARD_INIT=y
+CONFIG_HUSH_PARSER=y
+CONFIG_CMD_MEMINFO=y
+CONFIG_CMD_MEMTEST=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_I2C=y
+CONFIG_CMD_CACHE=y
+CONFIG_CMD_TIME=y
+CONFIG_CMD_PMIC=y
+CONFIG_CMD_REGULATOR=y
+CONFIG_CMD_FS_GENERIC=y
+CONFIG_CMD_UBI=y
+CONFIG_EFI_PARTITION=y
+# CONFIG_SPL_EFI_PARTITION is not set
+CONFIG_OF_CONTROL=y
+CONFIG_ENV_IS_IN_MMC=y
+CONFIG_IMX8M_LPDDR4=y
+CONFIG_SAVED_DRAM_TIMING_BASE=0x40000000
+CONFIG_DM_GPIO=y
+CONFIG_DM_I2C=y
+CONFIG_SYS_I2C_MXC=y
+CONFIG_DM_MMC=y
+CONFIG_PHY_GIGE=y
+CONFIG_DM_PMIC=y
+CONFIG_DM_PMIC_PFUZE100=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_PFUZE100=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
+CONFIG_DM_THERMAL=y
+CONFIG_NXP_TMU=y
+CONFIG_FS_EXT4=y
+CONFIG_EXT4_WRITE=y
+CONFIG_FS_FAT=y
+CONFIG_FAT_WRITE=y
diff --git a/include/configs/tqma8mx.h b/include/configs/tqma8mx.h
index 474b87d2c0..b87f5817f4 100644
--- a/include/configs/tqma8mx.h
+++ b/include/configs/tqma8mx.h
@@ -26,6 +26,7 @@
 #endif
 
 #ifdef CONFIG_SPL_BUILD
+/* #define CONFIG_SPL_ENV_SUPPORT */
 /*#define CONFIG_ENABLE_DDR_TRAINING_DEBUG*/
 #define CONFIG_SPL_WATCHDOG_SUPPORT
 #define CONFIG_SPL_DRIVERS_MISC_SUPPORT
@@ -243,9 +244,16 @@
 /* Size of malloc() pool */
 #define CONFIG_SYS_MALLOC_LEN		((CONFIG_ENV_SIZE + (2*1024) + (16*1024)) * 1024)
 
-#define CONFIG_SYS_SDRAM_BASE           0x40000000
-#define PHYS_SDRAM                      0x40000000
-#define PHYS_SDRAM_SIZE			0x40000000 /* 1GB DDR */
+#define CONFIG_SYS_SDRAM_BASE		0x40000000
+#define PHYS_SDRAM			0x40000000
+
+#if defined(CONFIG_TQMA8MX_2G_MICRON) || defined(CONFIG_TQMA8MX_2G_SAMSUNG)
+#define PHYS_SDRAM_SIZE			0x80000000 /* 2GB LPDDR4 */
+#else
+#define PHYS_SDRAM_SIZE			0x40000000 /* 1GB LPDDR4 */
+#error
+#endif
+
 #define CONFIG_NR_DRAM_BANKS		1
 
 #define CONFIG_SYS_MEMTEST_START	(CONFIG_SYS_SDRAM_BASE + SZ_128M)
